R/LSBuffonsneedlesimulation.R

```diff
@@ -0,0 +1,279 @@
+#
+# Copyright (C) 2019 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+LSBuffonsneedlesimulation<- function(jaspResults, dataset, options, state = NULL){
+
+  
+  # simulation results
+  #simulateResults <- simulate_throws(options[["n"]], options[["length"]])
+  
+  
+  if(is.null(jaspResults[["simulateResults"]])){ #test whether the state is empty
+    
+    # if empty, create a new state
+    simulateResults <- createJaspState()
+    simulateResults$dependOn(c("n", "length")) #, "a", "b", "options[["length"]]", "options[["CI"]]"))
+    jaspResults[["simulateResults"]] <- simulateResults
+    
+    jaspResults[["simulateResults"]]$object <- simulateThrows(options[["n"]], options[["length"]])  
+    
+  }
```

Comment:
you could put that in a function as well to clean up the main analysis function, but it's not a big deal...

Reply:
I created another function for this.